internal/pkg/redis/client.go

```diff
@@ -37,6 +38,10 @@ type Client struct {
 	// Client used for functionality related to reading messages
 	subscribeClient RedisClient
 
+	// Used to avoid multiple subscriptions to the same topic
+	existingTopics map[string]bool
```

Comment:
use struct as the value of map to save memory: https://medium.com/easyread/golang-series-empty-struct-ed317e6d8600  as reference

Reply:
changed